plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/Maven2ContentValidator.java

```diff
@@ -54,15 +55,18 @@ public Maven2ContentValidator(final DefaultContentValidator defaultContentValida
   @Override
   public String determineContentType(boolean strictContentTypeValidation,
                                      Supplier<InputStream> contentSupplier,
+                                     @Nullable MimeRulesSource mimeRulesSource,
                                      @Nullable String contentName,
                                      @Nullable String declaredContentType) throws IOException
   {
     if (contentName != null) {
       if (contentName.endsWith(".pom")) {
-        // hint to MimeSupport it's actually XML file
-        return defaultContentValidator
-            .determineContentType(strictContentTypeValidation, contentSupplier, contentName + ".xml",
-                declaredContentType);
+        // TODO: this is due fact that Tika has glob "*.pom" extension enlisted at text/plain, and some non-clean
```

Comment:
Why all of these TODOs? Please extract to a Jira issue if we want to follow up rather than muddying up the code